src/net_processing.cpp

```diff
@@ -4608,11 +4609,11 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
     // MaybeSendPing may have marked peer for disconnection
     if (pto->fDisconnect) return true;
 
-    MaybeSendAddr(*pto, *peer, current_time);
-
     {
         LOCK(cs_main);
 
+        MaybeSendAddr(*pto, *peer, current_time);
```

Comment:
Slightly tend toward NACK. We shouldn't be broadening the scope of cs_main, but reduce it. Seems wrong to use a validation lock for sending p2p addresses on the network.

Reply:
Agree. Addressed in latest push.